src/app.js

```diff
@@ -1,7 +1,20 @@
 /* @flow */
 'use strict'
 
+// SIGINT and uncaughtException do not run cleanup
+// so we have an exitHandler to clean up the database connection
+function exitHandler (error) {
+  if (error) {
+    console.error('Exiting: ', error)
+    process.exit(1)
+  }
+  process.exit(0)
+}
+
 module.exports = require('./lib/app')
 if (!module.parent) {
+  process.on('SIGINT', exitHandler)
+  process.on('uncaughtException', exitHandler)
```

Comment:
I think maybe the confusion lies in the interface Node provides for trapping signals. It would have made more sense for Node to call this `process.trap('SIGINT', callback)` --

http://tldp.org/LDP/Bash-Beginners-Guide/html/sect_12_02.html
https://docs.python.org/3.5/library/signal.html#signal.signal